src/rez/pip.py

```diff
@@ -81,35 +86,73 @@ def find_pip(pip_version=None, python_version=None):
     """
     py_exe = None
     context = None
+    found_pip_version = None
+    valid_found = False
 
-    py_exe, pip_version, context = find_pip_from_context(
-        python_version,
-        pip_version=pip_version
-    )
-
-    if not py_exe:
-        py_exe, pip_version, context = find_pip_from_context(
-            python_version,
-            pip_version=pip_version or "latest"
-        )
+    for version in [pip_version, "latest"]:
+        try:
+            py_exe, found_pip_version, context = find_pip_from_context(
+                python_version, pip_version=version
+            )
+            valid_found = _check_found(py_exe, found_pip_version)
+            if valid_found:
+                break
+        except BuildError as error:
+            print_warning(str(error))
 
-    if not py_exe:
+    if not valid_found:
         import pip
-        pip_version = pip.__version__
+
+        found_pip_version = pip.__version__
         py_exe = sys.executable
-        print_warning(
-            "Found no pip in python and pip package; "
-            "falling back to pip installed in rez own virtualenv (version %s)",
-            pip_version
+        print_warning("Found no pip in any python and/or pip packages!")
+        print_warning("Falling back to pip installed in rez own virtualenv:")
+        logging_arguments = (
+            ("pip", found_pip_version, pip.__file__),
+            ("python", ".".join(map(str, sys.version_info[:3])), py_exe),
         )
+        for warn_args in logging_arguments:
+            print_warning("%10s: %s (%s)", *warn_args)
 
-    pip_major = pip_version.split('.')[0]
-    if int(pip_major) < 19:
-        raise RezSystemError("pip >= 19 is required! Please update your pip.")
+    if not _check_found(py_exe, found_pip_version, log_invalid=False):
+        message = "pip{specifier} is required! Please update your pip."
+        raise RezSystemError(message.format(specifier=PIP_SPECIFIER))
 
     return py_exe, context
 
 
+def find_python_from_package(context, name="python", default=None):
+    """Get Python executable from current resolved python/pip context.
+
+    Args:
+        context (ResolvedContext): Resolved context with Python and pip.
+        name (str): Name of the package for Python instead of "python".
+        default (str): Force a particular fallback path for Python executable.
+
+    Returns:
+        str or None: Path to Python executable, if any.
+    """
+    py_exe_path = default
+    context = context.copy()
+    context.append_sys_path = False  # GitHub nerdvegas/rez/pull/826
+
+    py_packages = (v for v in context.resolved_packages if v.name == name)
+    python_package = next(py_packages)
+
+    if platform_.name == "windows" and python_package.version < Version("3"):
+        py_exe_path = context.which("python")  # GitHub nerdvegas/rez/pull/798
```

Comment:
confusing comment: 798 is an issue not a pr

Reply:
https://github.com/nerdvegas/rez/pull/798 is a pr which fixes issue https://github.com/nerdvegas/rez/issues/796